Show Operation Description Below Connected URL

Jatin2020-24 opened the pull request:
Fixes: #1281 
Fixes: #1286

Jatin2020-24 committed:
operation_description_below_connected_url

Jatin2020-24 committed:
minor_fixes

ReimarBauer commented on mslib/msui/qt5/ui_mainwindow.py:
```diff
@@ -287,3 +292,13 @@ def retranslateUi(self, MSSMainWindow):
         self.actionSearch.setText(_translate("MSSMainWindow", "Search"))
         self.actionSearch.setToolTip(_translate("MSSMainWindow", "Search for interactive text in the UI"))
         self.actionSearch.setShortcut(_translate("MSSMainWindow", "Ctrl+F"))
+
```
I guess this is created by the -x flag of pyuic5?
We don't need this extra code to test and display the class.

Jatin2020-24 replied:
Removed extra code.

Jatin2020-24 committed:
removed_extra_code_to_test_class

ReimarBauer commented on mslib/msui/qt5/ui_mainwindow.py:
```diff
@@ -241,10 +249,7 @@ def retranslateUi(self, MSSMainWindow):
-        self.listFlightTracks.setToolTip(_translate("MSSMainWindow", "List of open flight tracks.\n"
-"Double-click a flight track to activate it.\n"
-"Save a flight track to name it."))
+        self.activeOperationDesc.setText(_translate("MSSMainWindow", "Select Operation to View Description."))
```
@Jatin2020-24 I tried to find where this text is shown. When and where is it?

Jatin2020-24 replied:
if we add command `self.ui.activeOperationDesc.setHidden(False)` in `after_login` function then this text will be visible.
It will be good idea to show this text immediately after login, so user can easily understand how to see operation description.
![InkedScreenshot (318)_LI](https://user-images.githubusercontent.com/72596619/138241049-00a32887-5600-4ddf-80d9-acdddaad6770.jpg)

ReimarBauer replied:
that's a good idea, please do so.

ReimarBauer reviewed (approved):
see a question in the comment.

I have looked on the last used descriptions of the CIRRUS campaign. All used fits to the line.
We can change this when someone has a usecase for much more words.

ReimarBauer reviewed (changes requested):
a small change as suggested :)

ReimarBauer commented:
(I tried there the feature)

found another problem.

with logout the description must be cleared. Currently after the next login without selection the old text is shown again

![oldinfo](https://user-images.githubusercontent.com/278728/138325334-dbffdb00-949c-4747-8952-c409b68cfc49.gif)

Jatin2020-24 committed:
show_long_operation_description

- Tooltip extended on "operation-list" by "double-click" line.
- make compatible to show long descriptions.

Jatin2020-24 committed:
fixed_flake8

Jatin2020-24 commented:
now long descriptions (more than 95 chars) can be viewed from operations menu and for short desc. (less than 95 chars) can be viewed either from MSColab main window or in operation menu.
Tooltip has also been extended by "double-click" line.
![new](https://user-images.githubusercontent.com/72596619/138481341-2e380980-308d-47f2-9715-4901ffb75650.gif)

ReimarBauer reviewed (commented):
We have to take care on users which are stressed and so I think we should write 

Double click a operation to activate and view its description.

That is usually what a user won't know, how to get it activated.

Then it is "viewing description" an addon to the tooltip which is given from "Flight Tracks"

ReimarBauer commented:
Somehow I have the same problem as you showed in the gif too, it does not react on my first "ok" click. Also a click on the corner "x" does not destroy it on first attempt. 

interestingly sometimes it works perfect.

ReimarBauer commented on mslib/msui/mscolab.py:
```diff
@@ -1243,8 +1258,21 @@ def set_active_op_id(self, item):
             self.waypoints_model = None
 
+            # Show operation description
+            self.error_dialog = QtWidgets.QErrorMessage()
```
in this case it is no "error" likly information, or description.

please have a look if we can use too:
QtWidgets.QMessageBox.information

Jatin2020-24 replied:
changed error diaog to QMessage.information.

But, it is also not closing on single click.

ReimarBauer replied:
I created another issue to review this. It can be also closing and immeaditly reopen again.
But that is independent from this feature.

ReimarBauer reviewed (changes requested):
see comments.

if the closing of the window did not work as expected (many times of ok clicking) and there is no solution yet please add a # ToDo review closing of widget to that definition.

Jatin2020-24 committed:
added_QMessageBox

Jatin2020-24 committed:
fixed_flake8

Jatin2020-24 committed:
comment_updated

ReimarBauer reviewed (approved):
Thx
